lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -433,12 +453,14 @@ using FEA = FunctionEntrypointAnalyzer;
 FEA::FunctionEntrypointAnalyzer(llvm::Module &M,
                                 GeneratedCodeBasicInfo *GCBI,
                                 ArrayRef<GlobalVariable *> ABIRegs,
+                                BasicBlockQueue *EntrypointsQueue,
                                 FunctionAnalysisResults &Oracle,
-                                const TupleTree<model::Binary> &Binary) :
+                                TupleTree<model::Binary> &Binary) :
```

Comment:
`-collect-cfg` must work with a read-only model.
Maybe you can make this `const` and add a non-`const` `TupleTree<model::Binary> &Binary` argument to `finalizeModel` (which should probably be renamed to `commitToModel`)?